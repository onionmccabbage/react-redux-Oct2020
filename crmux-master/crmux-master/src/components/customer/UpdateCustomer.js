import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";

function UpdateCustomer(props) {
  const [customer, setCustomer] = useState(props.customer);

  const handleInputChange = (event) => {
    const { name, value } = event.target;

    setCustomer({ ...customer, [name]: value });
  };

  useEffect(() => {
    return () => {
      setCustomer(props.customer);
    };
  }, []);
  return (
    <form
      className="customerMaintenance"
      onSubmit={(event) => {
        event.preventDefault();
        props.onSubmitHandler(customer);
      }}
    >
      <label for="firstName">First Name</label>
      <input
        type="text"
        id="firstName"
        name="firstName"
        value={customer.firstName}
        onChange={handleInputChange}
        required
      />
      <label for="lastName">Last Name</label>
      <input
        type="text"
        id="lastName"
        name="lastName"
        value={customer.lastName}
        onChange={handleInputChange}
        required
      />
      <label for="emailAddress">Email Address</label>
      <input
        type="email"
        id="emailAddress"
        name="emailAddress"
        value={customer.emailAddress}
        onChange={handleInputChange}
        required
      />
      <button className="button">Update Customer</button>
    </form>
  );
}

UpdateCustomer.propTypes = {
  onSubmitHandler: PropTypes.func.isRequired,
  customer: PropTypes.object.isRequired,
};

export default UpdateCustomer;
