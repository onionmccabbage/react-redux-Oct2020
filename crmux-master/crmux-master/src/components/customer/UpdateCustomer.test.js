import React from "react";
import renderer from "react-test-renderer";
import UpdateCustomer from "./UpdateCustomer";

describe("The Customer Update component", () => {
  it("renders as expected", () => {
    const customer = {
      firstName: "John",
      lastName: "Wick",
      emailAddress: "john.wick@continental.com",
    };
    const callback = () => {};
    const tree = renderer
      .create(
        <UpdateCustomer
          onSubmitHandler={callback}
          customer={customer}
        ></UpdateCustomer>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
