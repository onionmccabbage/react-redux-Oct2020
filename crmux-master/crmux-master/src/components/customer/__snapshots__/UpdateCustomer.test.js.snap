// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Customer Update component renders as expected 1`] = `
<form
  className="customerMaintenance"
  onSubmit={[Function]}
>
  <label
    for="firstName"
  >
    First Name
  </label>
  <input
    id="firstName"
    name="firstName"
    onChange={[Function]}
    required={true}
    type="text"
    value="John"
  />
  <label
    for="lastName"
  >
    Last Name
  </label>
  <input
    id="lastName"
    name="lastName"
    onChange={[Function]}
    required={true}
    type="text"
    value="Wick"
  />
  <label
    for="emailAddress"
  >
    Email Address
  </label>
  <input
    id="emailAddress"
    name="emailAddress"
    onChange={[Function]}
    required={true}
    type="email"
    value="john.wick@continental.com"
  />
  <button
    className="button"
  >
    UpdateCustomer
  </button>
</form>
`;

exports[`The ProductImage component renders as expected 1`] = `
<form
  className="customerMaintenance"
  onSubmit={[Function]}
>
  <label
    for="firstName"
  >
    First Name
  </label>
  <input
    id="firstName"
    name="firstName"
    onChange={[Function]}
    required={true}
    type="text"
    value="John"
  />
  <label
    for="lastName"
  >
    Last Name
  </label>
  <input
    id="lastName"
    name="lastName"
    onChange={[Function]}
    required={true}
    type="text"
    value="Wick"
  />
  <label
    for="emailAddress"
  >
    Email Address
  </label>
  <input
    id="emailAddress"
    name="emailAddress"
    onChange={[Function]}
    required={true}
    type="email"
    value="john.wick@continental.com"
  />
  <button
    className="button"
  >
    UpdateCustomer
  </button>
</form>
`;
